### PR TITLE
feature:modify supernode 1.0 bridge

### PR DESCRIPTION
**Pull request by yunfeiyanggzq**
Signed-off-by: yunfeiyangbuaa <yunfeiyang@buaa.edu.cn>

### Ⅰ. Describe what this PR did
modify the v1.0 bridge,We now use the 0.3 bridge and we will use the v1.0 bridge in the future.the v1.0 bridge hasn't been finished,I modify it in this pr
I do the following things:

1. make supernode url in the format `api/v1.*`
2. add some supernode v1.0 bridge 
3. modify previous  supernode v1.0 bridge

### Ⅱ. Does this pull request fix one issue?

refrence #1082 #1114

### CODE REVIEW

**Comment by Starnop** [comment]
@yunfeiyanggzq Could you please refer to #1082 issue. THX.

**Comment by yunfeiyanggzq** [comment]
> @yunfeiyanggzq Could you please refer to #1082 issue. THX.
Done

File: supernode/server/peer_bridge.go [hunk 1]
```diff
@@ -65,6 +70,7 @@ func (s *Server) getPeer(ctx context.Context, rw http.ResponseWriter, req *http.
 
 	peer, err := s.PeerMgr.Get(ctx, id)
 	if err != nil {
+		logrus.Errorf("failed to get peer info by peerID{%s},err: %v", id, err)
```
Comment (Starnop) [inline supernode/server/peer_bridge.go]: logrus.Errorf("failed to get peer info by peerID(%s): %v", id, err)

File: supernode/server/router.go [hunk 2]
```diff
@@ -18,13 +18,14 @@ package server
...
 import (
 	"context"
-	"encoding/json"
-	"net/http"
-	"net/http/pprof"
 
 	"github.com/dragonflyoss/Dragonfly/apis/types"
 	"github.com/dragonflyoss/Dragonfly/version"
 
+	"encoding/json"
```
Comment (Starnop) [inline supernode/server/router.go]: put it ahead and make all golang packages as a group

File: supernode/server/task_bridge.go [hunk 3]
```diff
@@ -18,10 +18,19 @@ package server
...
 import (
 	"context"
-	"net/http"
 	"strconv"
 
+	"github.com/dragonflyoss/Dragonfly/apis/types"
+	"github.com/dragonflyoss/Dragonfly/pkg/constants"
+	"github.com/dragonflyoss/Dragonfly/pkg/errortypes"
+
+	"encoding/json"
```
Comment (Starnop) [inline supernode/server/task_bridge.go]: ditto

File: supernode/server/router.go [hunk 4]
```diff
@@ -51,16 +52,20 @@ func initRoute(s *Server) *mux.Router {
 
 		// v1
 		// peer
-		{Method: http.MethodPost, Path: "/peers", HandlerFunc: s.registerPeer},
-		{Method: http.MethodDelete, Path: "/peers/{id}", HandlerFunc: s.deRegisterPeer},
-		{Method: http.MethodGet, Path: "/peers/{id}", HandlerFunc: s.getPeer},
-		{Method: http.MethodGet, Path: "/peers", HandlerFunc: s.listPeers},
+		{Method: http.MethodPost, Path: "/api/v1/peers", HandlerFunc: s.registerPeer},
```
Comment (Starnop) [inline supernode/server/router.go]: Could you please also update the swagger definition? THX.

File: supernode/server/task_bridge.go [hunk 5]
```diff
@@ -34,3 +43,41 @@ func (s *Server) deleteTask(ctx context.Context, rw http.ResponseWriter, req *ht
...
+		return errors.Wrap(errortypes.ErrInvalidValue, err.Error())
+	}
+
+	resp, err := s.TaskMgr.Register(ctx, request)
+	if err != nil {
+		logrus.Errorf("failed to register task %+v: %v", request, err)
+		return err
+	}
+	logrus.Debugf("success to register task %+v", request)
+	return EncodeResponse(rw, http.StatusOK, &types.ResultInfo{
```
Comment (Starnop) [inline supernode/server/task_bridge.go]: use `http.StatusCreated`  instead

In addition, why wrap the result with `types.ResultInfo`. And it's inconsistent with the definition of swagger.

File: client/interface.go [hunk 6]
```diff
@@ -25,9 +25,24 @@ import (
...
 }
 
 // PreheatAPIClient defines methods of Container client.
 type PreheatAPIClient interface {
 	PreheatCreate(ctx context.Context, config *types.PreheatCreateRequest) (*types.PreheatCreateResponse, error)
 }
+
+// PeerAPIClient defines methods of peer related client.
+type PeerAPIClient interface {
+	PeerCreate(ctx context.Context, config *types.PeerCreateRequest) (*types.PeerCreateResponse, error)
```
[the remark this answers is not in the document]
Comment (yunfeiyanggzq) [reply client/interface.go]: emmm.can you give an example,I can't understand what do you mean

File: supernode/server/router.go (hunk 4 above)
Comment (yunfeiyanggzq) [reply supernode/server/router.go]: emmm,do u mean the comment?

File: supernode/server/task_bridge.go (hunk 5 above)
Comment (yunfeiyanggzq) [reply supernode/server/task_bridge.go]: > In addition, why wrap the result with `types.ResultInfo`. And it's inconsistent with the definition of swagger.
I do with the response just like the v0.3 bridge.all the  post response do as the format.and what `And it's inconsistent with the definition of swagger.` mean?

**Comment by allencloud** [comment]
I think you did a lot of change in this pull request. But you add little in PR description. Please describe the change in a list. Otherwise, I would have lots of confusions, why to change golang version, why to add interface definition functions? .....  @yunfeiyanggzq

**Comment by yunfeiyanggzq** [comment]
> I think you did a lot of change in this pull request. But you add little in PR description. Please describe the change in a list. Otherwise, I would have lots of confusions, why to change golang version, why to add interface definition functions? ..... @yunfeiyanggzq
> 
> In addition, linter CI fails:
> 
> ```
> ./hack/golangci-lint.sh
> Begin to check gocode with golangci-lint locally
> Detected that golangci-lint has already been installed. Start linting...
> ERRO Running error: context loading failed: failed to load program with go/packages: go [list -e -json -compiled=true -test=true -export=false -deps=true -find=false -mod=readonly -- ./...]: exit status 1: go: updates to go.mod needed, disabled by -mod=readonly
>  
> Makefile:172: recipe for target 'golangci-lint' failed
> make: *** [golangci-lint] Error 3
> Exited with code 2
> ```

emmm.sorry,something goes wrong,I am working on it ,

**Comment by yunfeiyanggzq** [comment]
because the supernode need to do  more things  to move to version 1.0. I remove the register task api function refrence #1118 . @Starnop  please review this pr again

**Comment by Starnop** [comment]
> make supernode url in the format api/v1.*
add some supernode v1.0 bridge
modify previous supernode v1.0 bridge

Could you please update your plan in the issue #1082.

**Comment by Starnop** [comment]
LGTM.

**Comment by yunfeiyanggzq** [comment]
> > make supernode url in the format api/v1.*
> > add some supernode v1.0 bridge
> > modify previous supernode v1.0 bridge
> 
> Could you please update your plan in the issue #1082.

**Pull request merged**